.github/workflows/ci.yml

```diff
@@ -0,0 +1,63 @@
+name: Test and lint
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version:
+          ["pypy-3.6", "pypy-3.7", "3.6", "3.7", "3.8", "3.9", "3.10-dev"]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install needed dependencies
+        run: |
+          pip install tox tox-gh-actions
+
+      - name: Run tests with tox
+        run: tox
+
+  lint:
+    runs-on: ubuntu-latest
+    env:
+      TARGETS: src tests
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Install Poetry
+        run: pip install poetry
+
+      - name: Install dependencies
+        run: |
+          poetry config virtualenvs.in-project true
```

Comment:
what's that line?

Reply:
With this config poetry creates virtual environment in the current folder. Otherwise it would go to the user's home directory.